registration-processor/core-processor/registration-processor-abis-handler-stage/src/main/java/io/mosip/registration/processor/abis/handler/stage/AbisHandlerStage.java

```diff
@@ -1,598 +1,681 @@
+			ObjectMapper mapper = new ObjectMapper();
+			ShareableAttributes shareableAttributes = mapper.readValue(mapper.writeValueAsString(attributes.get(0)),
+					ShareableAttributes.class);
+			for (Source source : shareableAttributes.getSource()) {
+				List<Filter> filterList = source.getFilter();
+				if (filterList != null && !filterList.isEmpty()) {
+
+					filterList.forEach(filter -> {
+						if (filter.getSubType() != null && !filter.getSubType().isEmpty()) {
+							typeAndSubTypeMap.put(filter.getType(), filter.getSubType());
+						} else {
+							typeAndSubTypeMap.put(filter.getType(), null);
+						}
+					});
+				}
+			}
+		}
+		return typeAndSubTypeMap;
+		
+	}
+
+	private void validateBiometricRecord(BiometricRecord biometricRecord, List<String> modalities,
+			Map<String, List<String>> biometricModalitySegmentsMap)
+			throws BiometricRecordValidationException, JsonParseException, JsonMappingException, IOException {
+		if (modalities == null || modalities.isEmpty()) {
+			throw new BiometricRecordValidationException(PlatformErrorMessages.RPR_DATASHARE_MODALITIES_EMPTY.getCode(),PlatformErrorMessages.RPR_DATASHARE_MODALITIES_EMPTY.getMessage());
+		}
+		if (biometricRecord == null || biometricRecord.getSegments() == null
+				|| biometricRecord.getSegments().isEmpty()) {
+			throw new BiometricRecordValidationException(PlatformErrorMessages.RPR_NO_BIOMETRICS_FOUND_WITH_DATASHARE.getCode(),PlatformErrorMessages.RPR_NO_BIOMETRICS_FOUND_WITH_DATASHARE.getMessage());
+		}
+
+		boolean isBioFound = false;
+		for (String biometricSegment : biometricModalitySegmentsMap.keySet()) {
+			if (!modalities.contains(biometricSegment)) {
+				throw new BiometricRecordValidationException(PlatformErrorMessages.RPR_BIOMETRIC_SEGMENT_NOT_CONFIGURED_FOR_MODALITY.getCode(),String.format(PlatformErrorMessages.RPR_BIOMETRIC_SEGMENT_NOT_CONFIGURED_FOR_MODALITY.getMessage(),biometricSegment));
+			}
+			for (String segment : biometricModalitySegmentsMap.get(biometricSegment)) {
+				Optional<BIR> optionalBIR = Optional.empty();
+				if (segment.equalsIgnoreCase("Face")) {
+					optionalBIR = biometricRecord.getSegments().stream()
+							.filter(bir -> bir.getBdbInfo().getType() != null
+									&& bir.getBdbInfo().getType().get(0).equals(BiometricType.FACE))
+							.findFirst();
+				} else {
+					String[] segmentArray = segment.split(" ");
+					optionalBIR = biometricRecord.getSegments().stream().filter(bir -> bir.getBdbInfo()
+							.getSubtype() != null && bir.getBdbInfo().getSubtype().size() == segmentArray.length
+									? (bir.getBdbInfo().getSubtype().get(0).equalsIgnoreCase(segmentArray[0])
+											&& (segmentArray.length == 2
+													? bir.getBdbInfo().getSubtype().get(1)
+															.equalsIgnoreCase(segmentArray[1])
+													: true))
+									: false)
+							.findFirst();
+				}
+				if (optionalBIR.isPresent()) {
+					BIR bir = optionalBIR.get();
+					if (bir.getBdb() != null) {
+						isBioFound = true;

```

Comment:
Can't we simply return from here and avoid using isBioFound variable ?

Reply:
If return from here then method signature need to change and exception need to through if it is false ,didnt find out one match segment

Reply:
Fixed